### DISCUSSION TITLE
qute-bitwarden userscript fails when logging into GitHub

### DISCUSSION THREAD

**Discussion by murchu27** [qutebrowser/qutebrowser#5950] [Q&A]
**Version info**:
qutebrowser v1.14.0
Git commit:
Backend: QtWebEngine (Chromium 80.0.3987.163)
Qt: 5.15.0

Platform: Manjaro Linux

**Does the bug happen if you start with `--temp-basedir`?**:
Yes

**Description**
Trying to run the `qute-bitwarden` userscript on the GitHub login screen, using the command:

    spawn --userscript qute-bitwarden

but I'm getting the following error, and login information doesn't get added:

```
Process stderr:
Traceback (most recent call last):
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 302, in <module>
    sys.exit(main(arguments))
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 240, in main
    choices = ['{:s} | {:s}'.format(c['name'], c['login']['username']) for c in candidates]
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 240, in <listcomp>
    choices = ['{:s} | {:s}'.format(c['name'], c['login']['username']) for c in candidates]
TypeError: unsupported format string passed to NoneType.__format__
```

I've tried using the script with other websites, e.g., Facebook, and that works, so I would presume my setup is fine. 

I've also tried using the Bitwarden extension on Firefox to fill in my GitHub details, and that works, so I don't think there's a problem with the vault/credentials.

Any idea what's up here?

**Comment by The-Compiler** [comment]
That error (`TypeError: unsupported format string passed to NoneType.__format__`) happens when `None` is passed to `.format(...)`. Thus, either `c['name']` or `c['login']['username']` is probably `None` here. I have no idea why that'd be, though... 

Maybe @JonnyHaystack or @typerslow who have contributed to that script have an idea?

**Comment by typerslow** [comment]
Since the Firefox extension works, I am assuming that `name` is the problem here. Try manually changing it to something else, does it still give the same error?

**Comment by JonnyHaystack** [comment]
Sorry I'm late to the party, I usually reply to stuff from mobile and discussions don't seem to work in FastHub yet..

Could it be that you haven't synced your vault with bitwarden-cli recently? You have to manually sync it. I launch the rofi-bitwarden script (not the qutebrowser one) and use Alt + R to sync the vault.
